client/src/main/java/io/split/storages/pluggable/adapters/UserCustomSplitAdapterConsumer.java

```diff
@@ -94,8 +96,20 @@ public List<String> splitNames() {
 
     @Override
     public Map<String, HashSet<String>> getNamesByFlagSets(List<String> flagSets) {
-        //Todo implement
-        return null;
+        Map<String, HashSet<String>> toReturn = new HashMap<>();
+        try {
+            UserPipelineWrapper pipelineExecution = _userStorageWrapper.pipeline();
+            for (String set: flagSets) {
```

Comment:
Edge case of course but you could just return `toReturn` if `flagSets` is `null`, to avoid the exception.
```suggestion
        try {
            if (flagSets == null) {
                return toReturn;
            }
            UserPipelineWrapper pipelineExecution = _userStorageWrapper.pipeline();
            for (String set: flagSets) {
```

Reply:
Updated.